client/src/main/java/org/apache/celeborn/client/read/WorkerPartitionReader.java

```diff
@@ -176,7 +186,8 @@ public ByteBuf next() throws IOException, InterruptedException {
       throw e;
     }
     returnedChunks++;
-    return chunk;
+    chunkIdsAlreadyReturned.add(chunk.getLeft());
```

Comment:
Only checkpoint when isCheckpointEnabled

Reply:
With the new way of checkpointing, this is now handled